File: DSA/link_list.c
#include <stdio.h>
#include <stdlib.h>
typedef struct node
{
    int data;
    struct node *next;
} node;

void display(node *head)
{
    node *temp = head;
    while (temp != 0)
    {
        printf("%d\t", temp->data);
        temp = temp->next;
    }
}
node *ins_atLast(node *head)
{
    node *temp = head, *new_node;
    new_node = (node *)malloc(sizeof(node));
    int num;
    printf("Enter the number: ");
    scanf("%d", &num);
    new_node->next=0;
    new_node->data=num;
    if (head == 0)
    {
        head = new_node;
    }
    else
    {

        while (temp->next != 0)
        {
            temp = temp->next;
        }
        temp->next = new_node;
    }
    return head;
}
void main()
{
    
    node *head = 0;
    head=ins_atLast(head);
    head=ins_atLast(head);
    head=ins_atLast(head);
    display(head);
}

File: DSA/Java/Link_list.java
public class Link_list {

    Node head;

    class Node {
        String data;
        Node next;

        Node(String data) {
            this.data = data;
            this.next = null;
        }
    }

    public void addFirst(String data) {
        Node new_node = new Node(data);
        if (head == null) {
            head = new_node;
            return;
        }
        new_node.next = head;
        head = new_node;
    }

    public void addLast(String data) {
        Node new_node = new Node(data);
        if (head == null) {
            head = new_node;
            return;
        }
        Node temp = head;
        while (temp.next != null) {
            temp = temp.next;
        }
        temp.next = new_node;

    }

    public void deleteLastNode(){
        if (head==null) {
            System.out.println("List is already empty");
            return;
        }
        Node temp=head;
        while (temp.next.next!=null) {
            temp=temp.next;
        }
        temp.next=null;

    }

    public void printList() {
        if (head == null) {
            System.out.println("List is empty");
        } else {
            Node temp = head;
            while (temp != null) {
                System.out.print(temp.data+"->");
                temp = temp.next;
            }
            System.out.println("null");
        }
    }

    public static void main(String[] args) {
        Link_list list = new Link_list();

        list.addLast("DJ");
        list.addLast("DJd");
        list.addLast("DJdd");
        list.addLast("DJddd");
        list.printList();
        list.deleteLastNode();
        list.printList();
        
    }
}